/* Copyright The OpenXLA Authors.
Copyright 2025 The ZKX Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package zkx;

enum ExecutionModeProto {
  EXECUTION_MODE_UNSPECIFIED = 0;
  EXECUTION_MODE_DEFAULT = 1;
  EXECUTION_MODE_SYNCHRONOUS = 2;
  EXECUTION_MODE_ASYNCHRONOUS = 3;
}

// Mirrors `xla::ExecuteOptions`.
message ExecuteOptionsProto {
  bool arguments_are_tupled = 1;
  bool untuple_result = 2;
  int32 launch_id = 3;
  bool strict_shape_checking = 4;
  bool use_major_to_minor_data_layout_for_callbacks = 8;
  ExecutionModeProto execution_mode = 6;
  repeated int32 non_donatable_input_indices = 7;
}
